lib/galaxy/tools/actions/__init__.py

```diff
@@ -903,13 +903,34 @@ def _wrapped_params(
         wrapped_params = WrappedParameters(trans, tool, incoming, input_datasets=input_datasets)
         return wrapped_params
 
-    def _get_on_text(self, inp_data):
+    def _get_on_text(self, inp_data, inp_dataset_collections):
         input_names = []
-        for data in reversed(list(inp_data.values())):
+        collection_names = []
+        for collections in inp_dataset_collections.values():
+            for dataset_collection, _ in collections:
+                if getattr(dataset_collection, "hid", None):
+                    collection_names.append(f"{dataset_collection.hid}")
+
+        for input_name, data in reversed(inp_data):
+            collection_hda_ids = set()
+            # output collection id and store included hda ids (to avoid extra inclusion in the list of datasets)
```

Comment:
I don't undestand this comment. What are you doing here ?

Reply:
Has been fixed